### PR TITLE
add more validation safety to loaders, CLI tool for validation

### PR DESCRIPTION
**Pull request by mcgibbon**
This PR adds some additional safety for loaders configuration, and a command-line tool to validate BatchesLoader configuration files.

Added public API:
- Added `validate_batches_config` command-line validation tool, which identifies formatting errors in BatchesLoader configuration files.
- Added `open_zarr` to registered mapper functions, and to top-level of loaders namespace
- Added post-init checks to loaders configuration classes which raise exceptions if invalid functions are used, instead of waiting until you attempt to load the data

Refactors of public API:
- fine res budget mapper no longer drops "time" from the dataset

Significant internal changes:
- Fixed bug in loaders tests where an error in a test would cause the context to fail to reset

- [x] Tests added

(Delete this for the commit message)
You are encouraged to check the PR against the [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--A4lKrs~xg7w5Gsb39N6JLNQoAg-IlsYffZgTwyKEylty7NhY).

### CODE REVIEW

File: external/loaders/loaders/_config.py [hunk 1]
```diff
@@ -48,6 +48,13 @@ def load_mapper(self) -> Mapper:
         mapping_func = mapper_functions[self.mapper_function]
         return mapping_func(self.data_path, **self.mapper_kwargs)
 
+    def __post_init__(self):
+        if self.mapper_function not in mapper_functions:
+            raise ValueError(
+                f"Invalid batches function {self.mapper_function}, "
```
Comment (AnnaKwa) [inline external/loaders/loaders/_config.py:54]:
```suggestion
                f"Invalid mapper function {self.mapper_function}, "
```

File: external/loaders/loaders/_config.py [hunk 2]
```diff
@@ -131,3 +145,10 @@ def load_batches(self, variables) -> Batches:
         """
         batches_function = batches_functions[self.batches_function]
         return batches_function(self.data_path, list(variables), **self.batches_kwargs,)
+
+    def __post_init__(self):
```
Comment (AnnaKwa) [inline external/loaders/loaders/_config.py:149]: Should this already have inherited its parent class `BatchesLoader`'s `__post_init__`?

File: external/loaders/loaders/mappers/_fine_resolution_budget.py [hunk 3]
```diff
@@ -434,7 +432,7 @@ def open_fine_res_apparent_sources(
         "delp": "pressure_thickness_of_atmospheric_layer",
     }
     dim_order = ("tile", "z", "y", "x")
-    drop_vars = ("step", "time")
+    drop_vars = ("step",)
```
Comment (AnnaKwa) [inline external/loaders/loaders/mappers/_fine_resolution_budget.py:435]: Did you want to also drop time in this mapper given that @brianhenn said it was alright?

**Review by AnnaKwa — APPROVED** [review]
LGTM! A couple minor comments.

File: external/loaders/loaders/mappers/_fine_resolution_budget.py (hunk 3 above)
Comment (mcgibbon) [reply external/loaders/loaders/mappers/_fine_resolution_budget.py:435]: Yes I want to stop dropping time in this mapper.

File: external/loaders/loaders/_config.py (hunk 2 above)
Comment (mcgibbon) [reply external/loaders/loaders/_config.py:149]: BatchesLoader does not have a `__post_init__` defined.

**Pull request merged**